lib/utils/config/definitions.js

```diff
@@ -1635,13 +1635,25 @@ define('progress', {
 define('provenance', {
   default: false,
   type: Boolean,
+  exclusive: ['provenance-file'],
```

Comment:
Do exclusive configs always need to be exclusive with each other? Is there such a thing as one-way exclusion? (the usage code seems to suggest that they must always be exclusive with each other).

Is there anything that could help ensure that the config are defined properly?

Reply:
They are exclusive with one another.  The three current applications are `browser` `provenance` and `expect-entries`.

We don't currently have a need for one-way exclusion, and if we ever did we'd have to revisit/refactor this.

Reply:
> Is there anything that could help ensure that the config are defined properly?

I'm not sure I understand this question.  The only problem this is trying to solve is having two mutually exclusive config items.

When originally thinking this out I thought that trying to force the two configs into one combined one was not a good idea, leaving it to the consumer to trust the exclusivity and look to either config accordingly (like this PR does w/ the provenance/provenance-file if statement.)

Reply:
Right. I agree they should be defined separately. I'm just thinking the code currently won't work right if, for example, flag A is exclusive to B but not the other way around, and if there isn't a way to make that easy to detect in future PRs etc.

Reply:
> > I'm just thinking the code currently won't work right if, for example, flag A is exclusive to B but not the other way around
> 
> I see what you're saying. That'd be a nice test to write like the ones we had to make sure each npm command had a certain attribute defined. It's not a currently a bug though because the current use case is that two flags are mutually exclusive. This means that either one can have the flag and we'll see the error state.

I think you're right that using the flags will still result in an error if only one flag is marked exclusive. But my reading is that, at least, the code for showing the help for the exclusive flags won't work properly as-is without both flags being marked as exclusive.

Reply:
Yep you're right.  I think we should address this when we go to add the other flags, since we can also then remove the ignored line of coverage from this PR handling the use case of imbalance.  I don't want what amounts to a docs linting tweak to hold this up.